<?xml version="1.0" encoding="UTF-8"?>
<!-- SESSION ID: 1527810458 -->
<launch>

  <!-- Calibration parameters -->
  <rosparam command="load" file="$(find ct_calibration)/config/camera_poses.yaml" />

  <!-- Network parameters -->
  <arg name="num_sensors"   default="3" />

  <arg name="sensor_0_id"   default="kinect01" />
  <arg name="sensor_0_name" default="$(arg sensor_0_id)" />

  <arg name="sensor_1_id"   default="kinect02" />
  <arg name="sensor_1_name" default="$(arg sensor_1_id)" />

  <arg name="sensor_2_id"   default="kinect03" />
  <arg name="sensor_2_name" default="$(arg sensor_2_id)" />

  <!-- Launching calibration -->
  <node pkg="opt_calibration" type="opt_define_reference_frame" name="opt_define_reference_frame" output="screen">
    <rosparam command="load" file="$(find ct_calibration)/config/camera_network.yaml" />

    <param name="num_sensors"           value="$(arg num_sensors)" />

    <param name="sensor_0/name"         value="/$(arg sensor_0_name)" />
    <param name="sensor_0/type"         value="pinhole_rgb" />
    <remap from="~sensor_0/image"       to="/$(arg sensor_0_name)/rgb_rect/image" />
    <remap from="~sensor_0/camera_info" to="/$(arg sensor_0_name)/rgb_rect/camera_info" />

    <param name="sensor_1/name"         value="/$(arg sensor_1_name)" />
    <param name="sensor_1/type"         value="pinhole_rgb" />
    <remap from="~sensor_1/image"       to="/$(arg sensor_1_name)/rgb_rect/image" />
    <remap from="~sensor_1/camera_info" to="/$(arg sensor_1_name)/rgb_rect/camera_info" />

    <param name="sensor_2/name"         value="/$(arg sensor_2_name)" />
    <param name="sensor_2/type"         value="pinhole_rgb" />
    <remap from="~sensor_2/image"       to="/$(arg sensor_2_name)/rgb_rect/image" />
    <remap from="~sensor_2/camera_info" to="/$(arg sensor_2_name)/rgb_rect/camera_info" />

  </node>

</launch>
